airbyte-cdk/java/airbyte-cdk/build.gradle

```diff
@@ -1,77 +1,84 @@
-plugins {
-    id 'java-library'
-    id 'maven-publish'
-}
-
-group 'io.airbyte'
-
-// Version is dynamically loaded from version.properties file.
-def props = new Properties()
-file("src/main/resources/version.properties").withInputStream { props.load(it) }
-version = props.getProperty('version')
-description = "Airbyte Connector Development Kit (CDK) for Java."
+allprojects {
+    apply plugin: 'java-library'
+    apply plugin: 'maven-publish'
+    apply plugin: 'airbyte-docker'
+    apply plugin: 'airbyte-java-cdk'
+    apply plugin: 'airbyte-integration-test-java'
+    apply plugin: 'airbyte-performance-test-java'
+    apply plugin: 'java-test-fixtures'
 
-dependencies {
-    testImplementation 'org.junit.jupiter:junit-jupiter:5.7.2'
+    group 'io.airbyte'
+    version = getCdkTargetVersion()
 }
 
-publishing {
-    publications {
-        maven(MavenPublication) {
-            groupId = 'io.airbyte'
-            artifactId = 'airbyte-cdk'
-            from components.java
+subprojects { subproject ->
+    def artifactBaseName = 'airbyte-cdk-' + subproject.name.replace('-feature', '')
+    // E.g. airbyte-cdk-core, airbyte-cdk-db-sources, airbyte-cdk-db-destinations, etc.
+
+    publishing {
+        publications {
+            main(MavenPublication) {
+                groupId = 'io.airbyte'
+                artifactId = artifactBaseName
+                from components.java
+            }
+            testFixtures(MavenPublication) {
+                groupId = 'io.airbyte'
+                artifactId = artifactBaseName + '-test-fixtures'
+                version = project.version
+                artifact subproject.tasks.testFixturesJar
+            }
         }
-    }
-    repositories {
-        maven {
-            name 'cloudrepo'
-            url 'https://airbyte.mycloudrepo.io/repositories/airbyte-public-jars'
-            credentials {
-                username System.getenv('CLOUDREPO_USER')
-                password System.getenv('CLOUDREPO_PASSWORD')
+        repositories {
+            maven {
+                name 'cloudrepo'
+                url 'https://airbyte.mycloudrepo.io/repositories/airbyte-public-jars'
+                credentials {
+                    username System.getenv('CLOUDREPO_USER')
+                    password System.getenv('CLOUDREPO_PASSWORD')
+                }
             }
         }
     }
+
+    project.configurations {
+        testImplementation.extendsFrom implementation
+        testFixturesImplementation.extendsFrom implementation
+        testFixturesRuntimeOnly.extendsFrom runtimeOnly
+    }
 }
 
-// Adds publishToMavenLocal as final command in the list of 'build' tasks.
-tasks.named('build').configure {
-    dependsOn tasks.named('publishToMavenLocal')
+description = "Airbyte Connector Development Kit (CDK) for Java."
+
+assemble {
+    dependsOn subprojects*.assemble
```

Comment:
Shouldn't we want to define dependencies lazily, i.e. `tasks.named('assemble').dependsOn ...` and friends? Also, do we really need to define these task dependencies explicitly at all? It seems unnecessary but I may be not seeing something.

Reply:
> Also, do we really need to define these task dependencies explicitly at all? It seems unnecessary but I may be not seeing something.

I double-checked with `--dry-run` to confirm. Without these mapped, the respective subproject tasks don't run.

> Shouldn't we want to define dependencies lazily, i.e. tasks.named('assemble').dependsOn ... and friends?

Yeah - happy to do so. In these cases, it should work either way, but no harm in making this change. Done ✅